src/app/(app)/attendance-report/actions/attendance-report.ts

```diff
@@ -27,67 +30,50 @@ export const getAttendanceReport = async () => {
     }
   };
 
-  // Get the first day of the month and today's date
-  const today = new Date();
-  const firstDayOfMonth = new Date(today.getFullYear(), today.getMonth(), 1);
-
-  // Total attendance days of current month
-  const totalAttendanceDays = (dateString: string) => {
-    const date = new Date(dateString);
-    return date >= firstDayOfMonth && date <= today;
+  const getTotalDaysInCurrentMonth = (): number => {
+    const today = new Date();
+    return new Date(today.getFullYear(), today.getMonth() + 1, 0).getDate();
   };
 
-  const uniqueMembers = Array.from(
-    new Set(attendanceData.map((attendance) => attendance.memberId))
-  );
+  const totalDaysInMonth = getTotalDaysInCurrentMonth();
+
+  const uniqueMembers = Array.from(new Set(attendanceData.map(({ memberId }) => memberId)));
 
   const monthlyAttendanceReport = uniqueMembers.map((memberId) => {
-    const memberAttendance = attendanceData.filter(
-      (attendance) =>
-        attendance.memberId === memberId &&
-        totalAttendanceDays(attendance.created_at)
-    );
+    const memberAttendance = attendanceData.filter(({ memberId: id }) => id === memberId);
 
     const presentDays = new Set(
       memberAttendance
-        .filter((attendance) => attendance.status === AttendanceStatus.Approve)
-        .map((attendance) => attendance.created_at?.split("T")[0])
+        .filter(({ status }) => status === AttendanceStatus.Approve)
+        .map(({ created_at }) => created_at.split("T")[0])
```

Comment:
@ghulamhafeez I think we should make use of the date function.
```suggestion
        .map(({ created_at }) => created_at.toLocaleDateString()
```

Reply:
Fixed